.github/workflows/main-build-and-deploy.yml

```diff
@@ -14,6 +14,22 @@ permissions:
   contents: read
 
 jobs:
+  run-tests-and-coverage:
+    name: Run tests & coverage
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v4
+      - uses: pnpm/action-setup@v3
+        with:
+          version: 8
+      - name: Install dependencies
+        run: pnpm install
+
+      - name: Run Tests & Coverage
+        run: pnpm test-and-coverage --outputFile=base-coverage/report.json
```

Comment:
can we get the test coverage reported to the Action as well
this way we can look at the latest build to see what the code coverage is up to